### PR TITLE
a dictionary should have key and value at the same time

### PR DESCRIPTION
**Pull request by whuan20011**
watch out when N < 5

### CODE REVIEW

File: combo/combo.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        master_dail1 = dail(b1, N, {})
+        master_dail2 = dail(b2, N, {})
+        master_dail3 = dail(b3, N, {})
+        count1 = same_number(farmer_dail1, master_dail1)
+        count2 = same_number(farmer_dail2, master_dail2)
+        count3 = same_number(farmer_dail3, master_dail3)
+        print >>fout, 250 - count1 * count2 * count3
+    else:
+        print >>fout, N * N * N
+def dail(num, N, dic):
```
Comment (smy20011) [inline combo/combo.py]: You can replace the logic of this function with mode
(num - 2 + N) mod N will always positive and within range [0, N)

File: combo/combo.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+TASK: combo
+"""
+def solution():
+    fin = open("combo.in", "r")
+    fout = open("combo.out", "w")
+    N = int(fin.readline().strip())
+    a1, a2, a3 = map(int, fin.readline().strip().split())
+    b1, b2, b3 = map(int, fin.readline().strip().split())
+    if N >= 5:
+        farmer_dail1 = dail(a1, N, {})
```
Comment (smy20011) [inline combo/combo.py]: You do not need pass the empty dict from function parameter. You can just create on in the dial function and return it.
Also, you can use map function over a list to reduce the boilerplate of your code.

File: combo/combo.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+"""
+ID: whuan2001
+LANG: PYTHON2
+TASK: combo
+"""
+def solution():
+    fin = open("combo.in", "r")
+    fout = open("combo.out", "w")
+    N = int(fin.readline().strip())
+    a1, a2, a3 = map(int, fin.readline().strip().split())
```
Comment (smy20011) [inline combo/combo.py]: You can save the result in a list.

File: combo/combo.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    b1, b2, b3 = map(int, fin.readline().strip().split())
+    if N >= 5:
+        farmer_dail1 = dail(a1, N, {})
+        farmer_dail2 = dail(a2, N, {})
+        farmer_dail3 = dail(a3, N, {})
+        master_dail1 = dail(b1, N, {})
+        master_dail2 = dail(b2, N, {})
+        master_dail3 = dail(b3, N, {})
+        count1 = same_number(farmer_dail1, master_dail1)
```
Comment (smy20011) [inline combo/combo.py]: count_overlap?

File: combo/combo.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ID: whuan2001
+LANG: PYTHON2
+TASK: combo
+"""
+def solution():
+    fin = open("combo.in", "r")
+    fout = open("combo.out", "w")
+    N = int(fin.readline().strip())
+    a1, a2, a3 = map(int, fin.readline().strip().split())
+    b1, b2, b3 = map(int, fin.readline().strip().split())
```
Comment (smy20011) [inline combo/combo.py]: Ditto

File: combo/combo.py (hunk 1 above)
Comment (whuan20011) [reply combo/combo.py]: I think we can not use mode, because when num == 1, num - 1 == 0  (0 + N) % N == 0

File: combo/combo.py (hunk 2 above)
Comment (whuan20011) [reply combo/combo.py]: I have deleted the empty dict and created dic in dail funtion.
How to use map function over a list?

File: combo/combo.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        count2 = same_number(farmer_dail2, master_dail2)
+        count3 = same_number(farmer_dail3, master_dail3)
+        print >>fout, 250 - count1 * count2 * count3
+    else:
+        print >>fout, N * N * N
+def dail(num, N):
+    dic = {}
+    dic[(num -1 - 1 + N) % N + 1] = True
+    dic[(num - 1 - 2 + N) % N + 1] = True
+    dic[num] = True
```
Comment (smy20011) [inline combo/combo.py:26]: should be num - 1

File: combo/combo.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+TASK: combo
+"""
+def solution():
+    fin = open("combo.in", "r")
+    fout = open("combo.out", "w")
+    N = int(fin.readline().strip())
+    a1, a2, a3 = map(int, fin.readline().strip().split())
+    b1, b2, b3 = map(int, fin.readline().strip().split())
+    if N >= 5:
+        farmer_dail1 = dail(a1, N)
```
Comment (smy20011) [inline combo/combo.py]: farmer_dial = map(lambda item: dail(item, N), arr)

Comment (smy20011) [reply combo/combo.py]:
```python
farmer_dial=[]
for item in arr:
    farmer_dial.append(dial(item, N))
```

File: combo/combo.py (hunk 4 above)
Comment (smy20011) [reply combo/combo.py]:
```python
overlap_count = 1
for farmer_poss, master_poss in zip(farmer_dial, master_dial):
    overlap_count *= same_number(farmer_poss, master_poss)
```

**Review by smy20011 — APPROVED** [review]

**Pull request merged**